json_serializable/lib/src/utils.dart

```diff
@@ -175,7 +175,19 @@ int _sortByLocation(FieldElement a, FieldElement b) {
 
 final _dartCoreObjectChecker = const TypeChecker.fromRuntime(Object);
 
-/// If a parameter is required to invoke the constructor,
+String genericClassArguments(ClassElement element, bool withConstraints) {
```

Comment:
we seem to be relying on `withConstraints` being non-null here because of  another method with the same name handling nullability. Maybe handle them both here and skip the second identically named wrapper method?

Reply:
Ack. Going to leave both. Having the helper in `generator_helper` is nice.

Did forward the null check to the inner method